pkg/controllers/clusterresourceplacement/resource_selector_test.go

```diff
@@ -881,18 +952,36 @@ func TestSortResource(t *testing.T) {
 		},
 		"should gather selected resources with CRD or Namespace in front  with order, second case": {
 			resources: []runtime.Object{crd1, crd2, deployment, namespace2, clusterRole},
-			want:      []runtime.Object{namespace2, crd2, crd1, clusterRole, deployment},
+			want:      []runtime.Object{namespace2, crd2, crd1, deployment, clusterRole},
+		},
+		"should gather selected resources with PersistentVolumeClaim in front with order": {
+			resources: []runtime.Object{deployment, pvc, namespace1, role},
+			want:      []runtime.Object{namespace1, pvc, deployment, role},
+		},
+		"should gather selected resources with Secret in front with order": {
+			resources: []runtime.Object{deployment, secret, namespace1, crd1, namespace2, role},
+			want:      []runtime.Object{namespace2, namespace1, crd1, secret, deployment, role},
+		},
+		"should gather selected resources with ConfigMap and Secret in front with order": {
+			resources: []runtime.Object{deployment, secret, namespace1, role, configMap, secret2},
+			want:      []runtime.Object{namespace1, configMap, secret2, secret, deployment, role},
+		},
+		"should gather selected all the resources with the right order": {
+			resources: []runtime.Object{configMap, deployment, role, crd1, pvc, secret2, clusterRole, secret, namespace1, namespace2, crd2},
+			want:      []runtime.Object{namespace2, namespace1, crd2, crd1, configMap, secret2, secret, pvc, deployment, clusterRole, role},
 		},
 	}
 
 	for testName, tt := range tests {
 		t.Run(testName, func(t *testing.T) {
-			sortResources(tt.resources)
-
-			// Check that the returned resources match the expected resources
-			diff := cmp.Diff(tt.want, tt.resources)
-			if diff != "" {
-				t.Errorf("sortResources() mismatch (-want +got):\n%s", diff)
+			// run many times to make sure it's stable
```

Comment:
Hi Ryan! `sort.Slice` does not guarantee stableness IIRC; shall we switch to `sort.SliceStable` instead?

Reply:
the difference between SliceStable is only if there are identical objects in the array which is not possible in our case as a ETCD guarantee.